iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/unique_ptr.inl

```diff
@@ -32,14 +34,14 @@ unique_ptr<T>::unique_ptr(T* const ptr, const function<void(T*)>& deleter) noexc
 }
```

Comment:
Not relevant for Axivion, but passing a `cxx::function` even by reference is a bad idea (mainly efficiency). However, it can probably not be fixed easily by just using `function_ref` as it will generally not be able to extract the callable and move it to the `function``(instead there would be at least two layers of type erasure indirection, which is costly).

So we leave it as is for now.

Reply:
@MatthiasKillat Agree, this should be addressed in a separate PR. Rather low prio, though.